common-npm-packages/packaging-common/util.ts

```diff
@@ -102,6 +116,86 @@ export enum LogType {
     error
 }
 
+export function getAccessToken(nugetFeedType:string, endpointInputKey:string, feedInputKey:string, packageToolType: PackageToolType) : string {
+    let accessToken: string = getSystemAccessToken();
+    if (nugetFeedType.toLowerCase() == "internal") {
+        let feed = getProjectAndFeedIdFromInputParam(feedInputKey);
+        accessToken = getUserAccessToken(tl.getInput(endpointInputKey), packageToolType, feed);
+    }
+
+    if (!accessToken) {
+        tl.debug(`Access token not set. Using System Access token.`);
+        return getSystemAccessToken();
+    }
+    return accessToken;
+}
+
+function getUserAccessToken(endpointName: string, packageToolType: PackageToolType, feed: any): string {
+    let token = "";
+    tl.debug("Checking if the token is already set.");
+
+    if(endpointName) {
+        tl.debug(`Checking if the endpoint ${endpointName} provided by user, can be used.`);
+        token = getAccessTokenFromServiceConnectionForInternalFeeds(endpointName);
+    }
+    else {
+        tl.debug("Checking if the credentials are set in the environment.");
+        token = getAccessTokenFromEnvironmentForInternalFeeds(feed, packageToolType);
+    }
+
+    return token;
+}
+
+function getAccessTokenFromServiceConnectionForInternalFeeds(endpointName: string): string {
+    let token: string = "";
+    let auth = tl.getEndpointAuthorization(endpointName, true);
+    let scheme = tl.getEndpointAuthorizationScheme(endpointName, true).toLowerCase();
+    switch(scheme)
+    {
+        case ("token"):
+            token = auth.parameters["apitoken"];
+            break;
+        default:
+            tl.warning("Invalid authentication type for internal feed. Use token based authentication.");
+            break;
+    }
+
+    return token;
+}
+
+function getAccessTokenFromEnvironmentForInternalFeeds(feed: any, packageToolType: PackageToolType) : string {
```

Comment:
In what case would we want to use environment variables as access tokens? Do we expect these variable to be set by the user?

Reply:
We are expecting the user to either use NugetAuthenticate or pass the service connection in the individual publish tasks. User need not set this in environment for the pipeline scenario.